File: editorial/chap06/python/answer_B36.py
# 入力
N, K = map(int, input().split())
S = input()

# ON となっているものの個数を数える
numON = 0
for i in range(N):
	if S[i] == '1':
		numON += 1

# 答えを出力
if numON%2 == K%2:
	print("Yes")
else:
	print("No")

File: editorial/chap06/cpp/answer_B36.cpp
#include <iostream>
#include <string>
using namespace std;
 
int N, K;
string S;
 
int main() {
	// 入力
	cin >> N >> K;
	cin >> S;
 
	// ON となっているものの個数を数える
	int numON = 0;
	for (int i = 0; i < N; i++) {
		if (S[i] == '1') numON += 1;
	}
 
	// 答えを出力
	if (numON % 2 == K % 2) cout << "Yes" << endl;
	else cout << "No" << endl;
	return 0;
}
